pkg/cmd/workflow/view/view.go

```diff
@@ -0,0 +1,253 @@
+		}
+		return utils.OpenInBrowser(workflowURL)
+	}
+
+	yaml, err := getWorkflowContent(client, repo, workflow)
+	if err != nil {
+		return fmt.Errorf("could not get workflow file content: %w", err)
+	}
+
+	theme := opts.IO.DetectTerminalTheme()
+	markdownStyle := markdown.GetStyle(theme)
+	if err := opts.IO.StartPager(); err != nil {
+		fmt.Fprintf(opts.IO.ErrOut, "starting pager failed: %v\n", err)
+	}
+	defer opts.IO.StopPager()
+
+	if !opts.Raw {
+		cs := opts.IO.ColorScheme()
+		out := opts.IO.Out
+
+		fmt.Fprintln(out, cs.Bold(workflow.Name))
+		fmt.Fprintf(out, "%s ID: %s", cs.Gray(workflow.Path), cs.Cyanf("%d", workflow.ID))
+
+		codeBlock := fmt.Sprintf("```yaml\n%s\n```", yaml)
+		rendered, err := markdown.RenderWithOpts(codeBlock, markdownStyle,
+			markdown.RenderOpts{
+				markdown.WithoutIndentation(),
+				markdown.WithoutWrap(),
+			})
+		if err != nil {
+			return err
+		}
+		_, err = fmt.Fprint(opts.IO.Out, rendered)
+		return err
+	}
+
+	if _, err := fmt.Fprint(opts.IO.Out, yaml); err != nil {
+		return err
+	}
+
+	if !strings.HasSuffix(yaml, "\n") {
+		_, err := fmt.Fprint(opts.IO.Out, "\n")
+		return err
+	}
+
+	return nil
+}
+
+func promptWorkflows(client *api.Client, repo ghrepo.Interface) (*shared.Workflow, error) {
+	workflows, err := shared.GetWorkflows(client, repo, 10)
+	if len(workflows) == 0 {
+		err = errors.New("no workflows are enabled")
+	}
+
+	if err != nil {
+		var httpErr api.HTTPError
+		if errors.As(err, &httpErr) && httpErr.StatusCode == 404 {
+			err = errors.New("no workflows are enabled")
+		}
+
+		return nil, fmt.Errorf("could not fetch workflows for %s: %w", ghrepo.FullName(repo), err)
+	}
+
+	filtered := []shared.Workflow{}
+	candidates := []string{}
+	for _, workflow := range workflows {
+		if !workflow.Disabled() {
+			filtered = append(filtered, workflow)
+			candidates = append(candidates, workflow.Name)
+		}
+	}
+
+	var selected int
+
+	err = prompt.SurveyAskOne(&survey.Select{
+		Message:  "Select a workflow",
+		Options:  candidates,
+		PageSize: 10,
+	}, &selected)
+	if err != nil {
+		return nil, err
+	}
+
+	return &filtered[selected], nil
+}
+
+func resolveWorkflow(io *iostreams.IOStreams, client *api.Client, repo ghrepo.Interface, workflowSelector string) (*shared.Workflow, error) {
+	if workflowSelector == "" {
+		return nil, errors.New("empty workflow selector")
+	}
+
+	idRE := regexp.MustCompile(`^\d+$`)
+
+	if idRE.MatchString(workflowSelector) {
+		workflow, err := getWorkflowByID(client, repo, workflowSelector)
+		if err != nil {
+			return nil, err
+		}
+		return workflow, nil
+	}
+
+	workflows, err := getWorkflowsByName(client, repo, workflowSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(workflows) == 0 {
+		return nil, fmt.Errorf("could not find any workflows named %s", workflowSelector)
+	}
+
+	if len(workflows) == 1 {
+		return &workflows[0], nil
+	}
+
+	if !io.CanPrompt() {
```

Comment:
This is just opinion but it feels like this function could be split into two. One for retrieving the workflow(s) and then a second one that can get called from `runView` if more than one workflow is retrieved.